pkg/server/commands/reverseexpand/reverse_expand.go

```diff
@@ -321,6 +326,7 @@ LoopOnEdges:
 					Relation: innerLoopEdge.TargetReference.GetRelation(),
 				},
 			}
+			atomic.AddUint32(resolutionMetadata.DispatchCount, 1)
```

Comment:
Why not put this as the first line of `execute()`?

Reply:
@miparnisari I spoke about this with @jon-whit so basically we want to add only the consecutive count of dispatches and not the first one i.e root. The difference is we have all the dispatch count higher by 1 but for the correctness of definition we are adding it right before dispatch happens instead of counting it from first.

Reply:
Am I understanding correctly that the initial (root) invocation of `execute` doesn't technically count as a dispatch? And because of this, we only want to count subsequent invocations?

If so, that makes technical sense but highlights the inconsistency of recursive behavior between all the operations. I really like the idea of a standard dispatch concept but we're not very close.

Reply:
>Am I understanding correctly that the initial (root) invocation of execute doesn't technically count as a dispatch?

Yes, that was my conversation with Jon

For me having it at top of execute means, even if we add a new call to execute, or a new recursion line, the one who is implementing doesn't need to remember to increase the dispatch count there and having all values off by just 1 didn't feel that big of a miscount but maybe it is. We do miss out on correctness.

Reply:
And that's fine. No further action is needed for this PR IMO. But just speaking generally, this highlights the inconsistency in behavior between execute and dispatch.

Reply:
i don't think being off by 1 is a big deal, but i won't block the pr for it

Reply:
It's not off by one if we're defining a list objects dispatch the same as a check dispatch.

Reply:
i didn't understand 🤔 

list objects dispatch = (reverse expand dispatches) + (Check dispatches)

Versus

list objects dispatch = (reverse expand dispatches - 1) + (Check dispatches)

i personally don't think it matters that much

Reply:
> If so, that makes technical sense but highlights the inconsistency of recursive behavior between all the operations. I really like the idea of a standard dispatch concept but we're not very close.

☝️ this +1. This is one of the reasons why I think it's important to establish a clear and consistent Dispatch pattern across the code base. Today we have:

* `ReverseExpand#execute`
* `Expand#Execute`
* `CheckResolver#ResolveCheck`
* etc..

The invocation patterns and recursion loopback behaviors differ between these as well, and they really don't need to. I proposed to @adriantam and @poovamraj as part of this lager body of work to introduce more uniform dispatch composition through better interface composition, namely:

```go
type Dispatcher interface {
  CheckDispatcher
  ReverseExpandDispatcher
  ListUsersDispatcher
  ExpandDispatcher
}

// replaces CheckResolver today
type CheckDispatcher interface {

  // replaces ResolveCheck today
  DispatchCheck(context.Context, *DispatchCheckRequest) (...)
}

// replaces ReverseExpandQuery today
type ReverseExpandDispatcher interface {

  // replaces ReverseExpandQuery#execute today
  DispatchReverseExpand(...)
}

etc..
```

The purpose of this is to force us to more consciously consider the dispatch semantics and pattern, because doing so makes everything more uniform in general.